ArgoTests/Tests/DecodedTests.swift

```diff
@@ -116,6 +116,35 @@ class DecodedTests: XCTestCase {
       XCTFail("Unexpected Error Occurred")
     }
   }
+
+  func testDecodedAlternativeWithSuccess_ReturnsSelf() {
+    let user: Decoded<User> = decode(JSONFromFile("user_with_email")!)
+    let user_error: Decoded<User> = decode(JSONFromFile("user_with_bad_type")!)
+    let alternative = user.alternative(user_error)
+
+    XCTAssertEqual(user.value?.id, alternative.value?.id)
+  }
+
+  func testDecodedAlternativeWithError_ReturnsOther() {
+    let user: Decoded<User> = decode(JSONFromFile("user_with_email")!)
+    let user_error: Decoded<User> = decode(JSONFromFile("user_with_bad_type")!)
+    let alternative = user_error.alternative(user)
+
+    XCTAssertEqual(user.value?.id, alternative.value?.id)
+  }
+
+  func testDecodedAlternativeWithTwoErrors_ReturnsOther() {
+    let user_error: Decoded<User> = decode(JSONFromFile("user_without_key")!)
+    let user_error2: Decoded<User> = decode(JSONFromFile("user_with_bad_type")!)
+    let alternative = user_error.alternative(user_error2)
+
+    switch alternative {
+    case let .Failure(.TypeMismatch(expected, actual)):
+      XCTAssert(alternative.description == "Failure(TypeMismatch(Expected \(expected), got \(actual)))")
```

Comment:
Line Length Violation: Line should be 100 characters or less: currently 104 characters (line_length)